napari/layers/_shapes_layer/model.py

```diff
@@ -0,0 +1,1521 @@
+                  self.mode == 'vertex_insert' or
+                  self.mode == 'vertex_remove'):
+                inds = np.isin(self.data._index, self.selected_shapes)
+                vertices = self.data._vertices[inds]
+                # If currently adding path don't show box over last vertex
+                if self.mode == 'add_path':
+                    vertices = vertices[:-1]
+
+                if self._hover_shape is None:
+                    face_color = 'white'
+                elif self._hover_vertex is None:
+                    face_color = 'white'
+                else:
+                    face_color = self._highlight_color
+                edge_color = self._highlight_color
+                self._node._subvisuals[0].set_data(vertices,
+                                                   size=self._vertex_size,
+                                                   face_color=face_color,
+                                                   edge_color=edge_color,
+                                                   edge_width=1.5,
+                                                   symbol='square',
+                                                   scaling=False)
+                self._node._subvisuals[1].set_data(pos=None, width=0)
+        elif self._is_selecting:
+            box = create_box(self._drag_box)
+            edge_color = self._highlight_color
+            self._node._subvisuals[0].set_data(np.empty((0, 2)), size=0)
+            self._node._subvisuals[1].set_data(pos=box[[0, 2, 4, 6, 0]],
+                                               color=edge_color, width=1.5)
+        else:
+            self._node._subvisuals[0].set_data(np.empty((0, 2)), size=0)
+            self._node._subvisuals[1].set_data(pos=None, width=0)
+
+    def _finish_drawing(self):
+        """Resets properties involed in shape drawing so that a new shape can
+        then be drawn
+        """
+        index = copy(self._moving_shape)
+        self._is_moving = False
+        self.selected_shapes = []
+        self._drag_start = None
+        self._drag_box = None
+        self._fixed_vertex = None
+        self._moving_shape = None
+        self._moving_vertex = None
+        self._hover_shape = None
+        self._hover_vertex = None
+        if self._is_creating is True and self.mode == 'add_path':
+            vertices = self.data._vertices[self.data._index == index]
+            if len(vertices) <= 2:
+                self.data.remove(index)
+            else:
+                self.data.edit(index, vertices[:-1])
+        if self._is_creating is True and self.mode == 'add_polygon':
+            vertices = self.data._vertices[self.data._index == index]
+            if len(vertices) <= 2:
+                self.data.remove(index)
+        self._is_creating = False
+        self.refresh()
+
+    def remove_selected(self):
+        """Removes any selected shapes
+        """
+        to_remove = np.sort(self.selected_shapes)[::-1]
```

Comment:
I tend to prefer using numpy for numpy things. In this case I think the most elegant form is:

```suggestion
        to_remove = sorted(self.selected_shapes, reverse=True)
```

Reply:
changed